sunkit_image/tests/test_granule.py

```diff
@@ -0,0 +1,211 @@
+import random
+
+import numpy as np
+import pytest
+
+import sunpy
+import sunpy.map as sm
+
+import sunkit_image.granule as granule
+
+
+@pytest.fixture(scope="session")
+def inputs():
+    import warnings
+
+    with warnings.catch_warnings():
+        data_map = sunpy.map.Map("sunkit_image/tests/granule_testdata.fits")
+    test_res = 0.016
+    test_method = "li"
+    return data_map, test_res, test_method
+
+
+def test_segment(inputs):
+    """
+    Unit tests for segment() function.
+    """
+
+    data_map, test_res, test_method = inputs
+
+    # -------- positive tests -------- :
+    segmented = granule.segment(data_map, test_method, test_res, True)
+
+    # Test 1: check that the returned type is correct
+    assert type(segmented) is sunpy.map.mapbase.GenericMap
+    # Test 2: get an array of pixel values and check it is not empty
+    initial_pix = sm.all_pixel_indices_from_map(data_map).value
+    seg_pixels = sm.all_pixel_indices_from_map(segmented).value
+    assert np.size(seg_pixels) > 0
+    # Test 3: check that the returned shape is unchanged

```

Comment:
```suggestion
```

Reply:
I apologize - I feel like for some reason on this platform its hard to see what some of the changes you have made are, probably especially when they involve removing or adding lines. What was the change here?

Reply:
To remove this line.